typescript/src/merchi.ts

```diff
@@ -0,0 +1,36 @@
+import { Entity } from './entity';
+import { Category } from './entities/categories';
+import { Product } from './entities/products';
+import { generateUUID } from './uuid';
+
+
+// the type of classes
+export interface Type<T, A extends any[]> extends
+   Function { new(...args: A): T; }
+
+function cloneClass<T, A extends []>(original: Type<T, A>): Type<T, A> {
+  // copy the constructor, but use the empty object as `this`
+  const copy = original.bind({});
+  // pick up any static members (this is shallow, the members are not copied)
+  Object.assign(copy, original);
+  return copy;
+}
+
+export class Merchi {
```

Comment:
why does this new class exist?

1. in order for an entities to create instances of their related entities, there needs to be some kind of repository where the related entity class can be looked up. otherwise every entity definition needs to import every other entity definition, which would be rather crazy
2. not used yet, but, very soon, i anticipate this class can be used to store semi-global "configuration" like current session token, current user, api secret tokens, and so on